### ISSUE TITLE
update example to glutin 0.31

### ISSUE THREAD

**Issue by fr-an-k** [grovesNL/glow#249]
glutin 0.28 API got scrapped due to Android issues.
It would be nice if the example gets a bump to the new glutin 0.30 api.
At least it was confusing for me as a newcomer.

**Comment by grovesNL** [comment]
Open to a PR to change this

**Comment by CarePackage17** [comment]
I've started dabbling with this [here](https://github.com/CarePackage17/glow/blob/glutin-update/examples/test/src/main.rs), though I've run into a bunch of problems:

- glutin's API changed *a lot*
- Many things moved to winit, so we need to include that 
- winit can work with other libs, so we need more glue code (glutin-winit)
- The flow of events is quite a bit different. make_current now happens inside the event loop, while in the sample a current context is expected to be present before we enter the event loop (all the GL resources are created there).
- This leads to a runtime panic like this:
```
Reading GL_VERSION failed. Make sure there is a valid GL context currently active.
stack backtrace:
   0: rust_begin_unwind
             at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:595:5
   1: core::panicking::panic_fmt
             at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/core/src/panicking.rs:67:14
   2: glow::native::Context::from_loader_function_cstr
             at /home/krzysztof/devel/glow/src/native.rs:56:13
   3: glow::native::Context::from_loader_function
             at /home/krzysztof/devel/glow/src/native.rs:108:9
   4: test::main
             at ./src/main.rs:63:22
```

Additionally, I'm not at all familiar with any of the libraries involved. Most of this is scraped/copy-pasted together from docs and sample code.
My main motivation is to make a little game, and wanting to use egui for UI and OpenGL ES for regular rendering.

I guess a way to fix this would be to move the GL resource creation somewhere else (similar to what the glutin window example does IIRC), but that would break the web and SDL2 samples (which is why I created a "new" sample).

I'd appreciate some help. :)

**Comment by grovesNL** [comment]
@CarePackage17 awesome, yeah we should probably split off the examples anyway. I helped somebody get started with glutin and glow a couple days ago with this sample https://github.com/grovesNL/nikeedev-rs-opengl/blob/help/src/main.rs in case it's helpful

**Comment by CarePackage17** [comment]
Your sample code looks a lot more readable, whereas the thing I cobbled together might work on more platforms (by virtue of the glutin sample having a bunch of special cases + raw-window-handle being a requirement on Windows IIUC).

I've updated my fork and it runs fine on Ubuntu 23.10 as of now. I want to test it on Windows as well, because it'd be nice to have that out of the box too.

I guess the main question is how portable do you want the sample to be? My impression with the original sample is that it should work across desktop and web...but then again, you also suggested splitting the samples, so having one of them not work on all platforms might not be bad/can be improved in the future.

Should I open a PR?

**Comment by fr-an-k** [comment]
Looks like you also modified glutin's example to use glow, which is what I did.

Ideally there would also be a quick start for Android added to glow; you'd pretty much just have to copy the rest of the example from glutin.

So far I got glow working on Android 8.1+.
I suspect cargo apk has an issue on Android 5.1, or at least the library won't load without any reason why; just so you know.

**Comment by grovesNL** [comment]
Fixed by #276
